src/blockchain/bitcoin/p2p/message/Cfilter.cpp

```diff
@@ -172,19 +169,13 @@ Cfilter::Cfilter(
 auto Cfilter::payload(AllocateOutput out) const noexcept -> bool
 {
     try {
-        const auto payload = [&] {
-            auto filter = [&] {
-                auto output = CompactSize(count_).Encode();
-                output.insert(output.end(), filter_.begin(), filter_.end());
-
-                return output;
-            }();
-            auto output = CompactSize(filter.size()).Encode();
-            output.reserve(output.size() + filter.size());
-            std::move(filter.begin(), filter.end(), std::back_inserter(output));
-
-            return output;
-        }();
+        auto filter = CompactSize(count_).Encode();
+        filter.insert(filter.end(), filter_.begin(), filter_.end());
+
+        auto payload = CompactSize(filter.size()).Encode();
+        payload.reserve(payload.size() + filter.size());
+        std::move(filter.begin(), filter.end(), std::back_inserter(payload));
```

Comment:
I am utterly confused by this line. Is it just me, or does it require a comment? Assuming, it does anything at all other than three evaluations with and without side effect?